SEAB-6002: Remove residual notebook preview verbiage/code

svonworl opened the pull request:
**Description**
This PR removes the "PREVIEW" label from the "Notebooks" search tab, and adjusts `WorkflowsComponent` to display an "Explore Notebooks" header that's linked to notebooks search.

~~As part of the work, we added an `entryTypeMetadata$` observable to `SessionQuery`.  It's a little weird, in that the existing `SessionQuery.entryType$` is a swagger object, and the new `SessionQuery.entryTypeMetadata$` is an openApi object, but this inconsistency will go away when we someday convert to 100% openApi.~~

The "notebooks" feature flag and most of the notebooks preview UI were removed in https://ucsc-cgl.atlassian.net/issues/SEAB-5568

**Review Instructions**
So to dockstore search and confirm that the "Notebooks" search tab does not have a "PREVIEW" label.  Click on a notebook, and confirm that the "Explore Notebooks" text in the header is linked to notebooks search.

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-6002

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone.

coverbeck commented:
> As part of the work, we added an entryTypeMetadata$ observable to SessionQuery. It's a little weird, in that the existing SessionQuery.entryType$ is a swagger object, and the new SessionQuery.entryTypeMetadata$ is an openApi object, but this inconsistency will go away when we someday convert to 100% openApi.

I'm just going to start looking at the code, but I'm confused by this comment; the UI should be 100% OpenAPI. The last traces of Swagger were removed in #1815; we don't generate Swagger client code in the UI any more.

svonworl commented:
> > As part of the work, we added an entryTypeMetadata$ observable to SessionQuery. It's a little weird, in that the existing SessionQuery.entryType$ is a swagger object, and the new SessionQuery.entryTypeMetadata$ is an openApi object, but this inconsistency will go away when we someday convert to 100% openApi.
> 
> I'm just going to start looking at the code, but I'm confused by this comment; the UI should be 100% OpenAPI. The last traces of Swagger were removed in #1815; we don't generate Swagger client code in the UI any more.

You are correct, I was confused, `EntryType` is a custom enum of our own creation, not Swagger.  It's still a weird little inconsistency, but not nearly as noteworthy. :)  I adjusted the PR description.
